agent/grpc/public/services/peerstream/stream_tracker.go

```diff
@@ -73,6 +75,18 @@ func (t *Tracker) StreamStatus(id string) (resp Status, found bool) {
 	return s.GetStatus(), true
 }
 
+// MutableStreamStatus gives a handle of the MutableStatus for a given peer ID
+func (t *Tracker) MutableStreamStatus(id string) (status *MutableStatus, err error) {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	s, ok := t.streams[id]
+	if !ok {
+		return nil, fmt.Errorf("did not find a stream for id: %q", id)
```

Comment:
Can this return a boolean instead of an error for consistency with `StreamStatus`?